server/src/main/java/io/crate/statistics/DistinctValuesSketch.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.statistics;
+
+import java.io.IOException;
+
+import org.apache.datasketches.memory.Memory;
+import org.apache.datasketches.theta.SetOperation;
+import org.apache.datasketches.theta.Sketch;
+import org.apache.datasketches.theta.Sketches;
+import org.apache.datasketches.theta.Union;
+import org.apache.datasketches.theta.UpdateSketch;
+import org.elasticsearch.common.io.stream.StreamInput;
+
+/**
+ * A streamable representation of a distinct values sketch
+ */
+public abstract class DistinctValuesSketch {
+
+    /**
+     * Add a new value to the sketch
+     */
+    public abstract void update(String v);
+
+    /**
+     * Merge this sketch with another
+     */
+    public abstract DistinctValuesSketch merge(DistinctValuesSketch other);
+
+    /**
+     * Get the internal sketch data structure
+     */
+    public abstract Sketch getSketch();
+
+    /**
+     * Creates a new empty sketch
+     */
+    public static DistinctValuesSketch newSketch() {
+        return new DistinctValuesSketch() {
+
+            final UpdateSketch distinctSketch = UpdateSketch.builder().build();
+
+            @Override
+            public void update(String v) {
+                distinctSketch.update(v);
```

Comment:
Is this okay to go with String only here? The `UpdateSketch` class has overloads for the different types.

Or is this also related  to the TODO about having more type specific sketches?

Reply:
We can specialise this when we get numeric sketches, yes.  But at the moment the input could be any object, so we need to convert it to something that works for any type.